src/training_hub/profiling/memory_estimator.py

```
@@ -288,6 +314,195 @@ def estimate(self) -> tuple[int, int, int]:
+    def _resolve_lora(self, lora_r: int, target_terms: list[str]):
+        """
+        Determine how many parameters will be in LoRA's low rank matrices
+        """
+        # Get the total dimensionality of weight matrices in this model
+        weight_size_total = 0
+
+        # Iterate through all layers...
+        for layer_name in self.model.state_dict().keys():
+            # Check to see if this current layer is a weight matrix
+            layer_data = self.model.state_dict()[layer_name]
+            if layer_data.dim() < 2 or layer_name.find('weight') <= -1: continue
```

Ruff (0.14.8)

Multiple statements on one line (colon)

(E701)